src/test/unit_tests/test_bucketspace_versioning.js

```diff
@@ -3192,6 +3220,626 @@ mocha.describe('List-objects', function() {
         });
         assert.equal(count, 6);
     });
+
+    mocha.it('list object versions - no objects in the bucket (versioning enabled)', async function() {
+        await s3_client.putBucketVersioning({ Bucket: bucket_name2, VersioningConfiguration: { MFADelete: 'Disabled', Status: 'Enabled' } });
+        const res = await s3_client.listObjectVersions({ Bucket: bucket_name2 });
+        assert.equal(res.$metadata.httpStatusCode, 200);
```

Comment:
I suggest checking the number of objects equals to 0

Reply:
same for the equivalent suspended case

Reply:
We don't have any versions yet (so the array is undefined), I added a check for that, thanks.